diff -ru old/regina-4.96/engine/data/census/CMakeLists.txt new/regina-4.96/engine/data/census/CMakeLists.txt
--- old/regina-4.96/engine/data/census/CMakeLists.txt	2014-08-31 11:10:33.000000000 +1000
+++ new/regina-4.96/engine/data/census/CMakeLists.txt	2014-08-31 11:51:46.000000000 +1000
@@ -2,7 +2,7 @@
 SET(CENSUSDATADIR ${PKGDATADIR}/data/census)
 
 INCLUDE_DIRECTORIES(${ENGINE_INCLUDES})
-ADD_EXECUTABLE(mkcensusdb mkcensusdb.cpp ../../utilities/zstream.cpp)
+ADD_EXECUTABLE(mkcensusdb mkcensusdb.cpp)
 TARGET_LINK_LIBRARIES(mkcensusdb ${ZLIB_LIBRARIES} ${TOKYOCABINET_LIBRARIES})
 
 SET(censusdata_DATA
diff -ru old/regina-4.96/engine/data/census/mkcensusdb.cpp new/regina-4.96/engine/data/census/mkcensusdb.cpp
--- old/regina-4.96/engine/data/census/mkcensusdb.cpp	2014-08-31 11:40:21.000000000 +1000
+++ new/regina-4.96/engine/data/census/mkcensusdb.cpp	2014-08-31 11:53:14.000000000 +1000
@@ -48,7 +48,10 @@
 #include <tcbdb.h>
 #include <tcutil.h>
 #endif
-#include "utilities/zstream.h"
+
+#include "regina-core.h"
+#define REGINA_API
+#include "../../utilities/zstream.cpp"
 
 #ifdef QDBM_AS_TOKYOCABINET
   #define DB_CLOSE(x) vlclose(x);
