diff --git a/python/angle/nanglestructurelist.cpp b/python/angle/nanglestructurelist.cpp
index 8e10210..f348d8d 100644
--- a/python/angle/nanglestructurelist.cpp
+++ b/python/angle/nanglestructurelist.cpp
@@ -34,6 +34,7 @@
 #include "progress/nprogresstracker.h"
 #include "triangulation/ntriangulation.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -96,5 +97,7 @@ void addNAngleStructureList() {
 
     implicitly_convertible<SafeHeldType<NAngleStructureList>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NAngleStructureList);
 }
 
diff --git a/python/dim2/dim2triangulation.cpp b/python/dim2/dim2triangulation.cpp
index 4283136..6ca5d1e 100644
--- a/python/dim2/dim2triangulation.cpp
+++ b/python/dim2/dim2triangulation.cpp
@@ -273,6 +273,8 @@ void addDim2Triangulation() {
         SafeHeldType<regina::NPacket> >();
     }
 
+    FIX_REGINA_BOOST_CONVERTERS(Dim2Triangulation);
+
     scope().attr("Dim2Triangulation") = scope().attr("Triangulation2");
 }
 
diff --git a/python/dim4/dim4triangulation.cpp b/python/dim4/dim4triangulation.cpp
index 995bacd..f9bea60 100644
--- a/python/dim4/dim4triangulation.cpp
+++ b/python/dim4/dim4triangulation.cpp
@@ -345,6 +345,8 @@ void addDim4Triangulation() {
         SafeHeldType<regina::NPacket> >();
     }
 
+    FIX_REGINA_BOOST_CONVERTERS(Dim4Triangulation);
+
     scope().attr("Dim4Triangulation") = scope().attr("Triangulation4");
 }
 
diff --git a/python/generic/triangulation-bindings.h b/python/generic/triangulation-bindings.h
index 77c3bd4..dad0d33 100644
--- a/python/generic/triangulation-bindings.h
+++ b/python/generic/triangulation-bindings.h
@@ -262,5 +262,7 @@ void addTriangulation(const char* name) {
 
     implicitly_convertible<SafeHeldType<Triangulation<dim>>,
         SafeHeldType<regina::NPacket>>();
+
+    FIX_REGINA_BOOST_CONVERTERS(Triangulation<dim>);
 }
 
diff --git a/python/helpers.h b/python/helpers.h
index 7bf59a4..9f11a7b 100644
--- a/python/helpers.h
+++ b/python/helpers.h
@@ -42,4 +42,13 @@
 #include "helpers/equality.h"
 #include "helpers/output.h"
 
+#if BOOST_VERSION >= 106000 && BOOST_VERSION < 106100
+    // Boost.Python 1.60 did not automatically register to_python converters
+    // for SafeHeldType.
+    #define FIX_REGINA_BOOST_CONVERTERS(T) \
+        register_ptr_to_python<regina::python::SafeHeldType<T>>()
+#else
+    #define FIX_REGINA_BOOST_CONVERTERS(T)
+#endif
+
 #endif
diff --git a/python/hypersurface/nnormalhypersurfacelist.cpp b/python/hypersurface/nnormalhypersurfacelist.cpp
index abd5c62..4bc8043 100644
--- a/python/hypersurface/nnormalhypersurfacelist.cpp
+++ b/python/hypersurface/nnormalhypersurfacelist.cpp
@@ -35,6 +35,7 @@
 #include "maths/nmatrixint.h"
 #include "progress/nprogresstracker.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -104,5 +105,7 @@ void addNNormalHypersurfaceList() {
 
     implicitly_convertible<SafeHeldType<NNormalHypersurfaceList>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NNormalHypersurfaceList);
 }
 
diff --git a/python/packet/ncontainer.cpp b/python/packet/ncontainer.cpp
index 8b8b06a..b6d7dea 100644
--- a/python/packet/ncontainer.cpp
+++ b/python/packet/ncontainer.cpp
@@ -32,6 +32,7 @@
 
 #include "packet/ncontainer.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -52,5 +53,7 @@ void addNContainer() {
 
     implicitly_convertible<SafeHeldType<NContainer>,
         SafeHeldType<regina::NPacket>>();
+
+    FIX_REGINA_BOOST_CONVERTERS(NContainer);
 }
 
diff --git a/python/packet/npacket.cpp b/python/packet/npacket.cpp
index ba0d94d..282aa6a 100644
--- a/python/packet/npacket.cpp
+++ b/python/packet/npacket.cpp
@@ -174,5 +174,6 @@ void addNPacket() {
 
     def("open", open_filename, return_value_policy<to_held_type<> >());
 
+    FIX_REGINA_BOOST_CONVERTERS(NPacket);
 }
 
diff --git a/python/packet/npdf.cpp b/python/packet/npdf.cpp
index 9db58b5..4ed0e0f 100644
--- a/python/packet/npdf.cpp
+++ b/python/packet/npdf.cpp
@@ -32,6 +32,7 @@
 
 #include "packet/npdf.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -59,5 +60,7 @@ void addNPDF() {
 
     implicitly_convertible<SafeHeldType<NPDF>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NPDF);
 }
 
diff --git a/python/packet/nscript.cpp b/python/packet/nscript.cpp
index 3f23e51..964f326 100644
--- a/python/packet/nscript.cpp
+++ b/python/packet/nscript.cpp
@@ -32,6 +32,7 @@
 
 #include "packet/nscript.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -89,5 +90,7 @@ void addNScript() {
 
     implicitly_convertible<SafeHeldType<NScript>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NScript);
 }
 
diff --git a/python/packet/ntext.cpp b/python/packet/ntext.cpp
index 9f0b6d7..f1820ff 100644
--- a/python/packet/ntext.cpp
+++ b/python/packet/ntext.cpp
@@ -32,6 +32,7 @@
 
 #include "packet/ntext.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -63,5 +64,7 @@ void addNText() {
 
     implicitly_convertible<SafeHeldType<NText>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NText);
 }
 
diff --git a/python/snappea/nsnappeatriangulation.cpp b/python/snappea/nsnappeatriangulation.cpp
index 679f4f9..38bbc52 100644
--- a/python/snappea/nsnappeatriangulation.cpp
+++ b/python/snappea/nsnappeatriangulation.cpp
@@ -169,5 +169,7 @@ void addNSnapPeaTriangulation() {
 
     implicitly_convertible<SafeHeldType<NSnapPeaTriangulation>,
         SafeHeldType<regina::NTriangulation> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NSnapPeaTriangulation);
 }
 
diff --git a/python/surfaces/nnormalsurfacelist.cpp b/python/surfaces/nnormalsurfacelist.cpp
index 11d5fdd..da46e93 100644
--- a/python/surfaces/nnormalsurfacelist.cpp
+++ b/python/surfaces/nnormalsurfacelist.cpp
@@ -35,6 +35,7 @@
 #include "surfaces/nnormalsurfacelist.h"
 #include "triangulation/ntriangulation.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -245,5 +246,7 @@ void addNNormalSurfaceList() {
 
     implicitly_convertible<SafeHeldType<NNormalSurfaceList>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NNormalSurfaceList);
 }
 
diff --git a/python/surfaces/nsurfacefilter.cpp b/python/surfaces/nsurfacefilter.cpp
index 848506e..6fde81a 100644
--- a/python/surfaces/nsurfacefilter.cpp
+++ b/python/surfaces/nsurfacefilter.cpp
@@ -33,6 +33,7 @@
 #include "surfaces/nnormalsurface.h"
 #include "surfaces/nsurfacefilter.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -59,5 +60,7 @@ void addNSurfaceFilter() {
 
     implicitly_convertible<SafeHeldType<NSurfaceFilter>,
         SafeHeldType<regina::NPacket> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NSurfaceFilter);
 }
 
diff --git a/python/surfaces/sfcombination.cpp b/python/surfaces/sfcombination.cpp
index 3a05ccc..96a0d67 100644
--- a/python/surfaces/sfcombination.cpp
+++ b/python/surfaces/sfcombination.cpp
@@ -32,6 +32,7 @@
 
 #include "surfaces/sfcombination.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -54,5 +55,7 @@ void addNSurfaceFilterCombination() {
 
     implicitly_convertible<SafeHeldType<NSurfaceFilterCombination>,
         SafeHeldType<regina::NSurfaceFilter> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NSurfaceFilterCombination);
 }
 
diff --git a/python/surfaces/sfproperties.cpp b/python/surfaces/sfproperties.cpp
index ad59189..cff6818 100644
--- a/python/surfaces/sfproperties.cpp
+++ b/python/surfaces/sfproperties.cpp
@@ -32,6 +32,7 @@
 
 #include "surfaces/sfproperties.h"
 #include "../safeheldtype.h"
+#include "../helpers.h"
 
 // Held type must be declared before boost/python.hpp
 #include <boost/python.hpp>
@@ -82,5 +83,7 @@ void addNSurfaceFilterProperties() {
 
     implicitly_convertible<SafeHeldType<NSurfaceFilterProperties>,
         SafeHeldType<regina::NSurfaceFilter> >();
+
+    FIX_REGINA_BOOST_CONVERTERS(NSurfaceFilterProperties);
 }
 
diff --git a/python/triangulation/ntriangulation.cpp b/python/triangulation/ntriangulation.cpp
index ee34716..b2525fd 100644
--- a/python/triangulation/ntriangulation.cpp
+++ b/python/triangulation/ntriangulation.cpp
@@ -471,6 +471,8 @@ void addNTriangulation() {
         SafeHeldType<regina::NPacket> >();
     }
 
+    FIX_REGINA_BOOST_CONVERTERS(NTriangulation);
+
     scope().attr("NTriangulation") = scope().attr("Triangulation3");
 }
 
